src/Driver/Driver.php

```diff
@@ -676,16 +685,25 @@ protected function getPDO(): PDO|PDOInterface
      * Creating a context for logging
      *
      * @param float $queryStart Query start time
+     * @param PDOStatement|PDOStatementInterface|null $statement Statement object
+     * @param iterable|null $parameters Query parameters
+     *
+     * @return array
      */
-    protected function defineLoggerContext(float $queryStart, PDOStatement|PDOStatementInterface|null $statement): array
+    protected function defineLoggerContext(float $queryStart, PDOStatement|PDOStatementInterface|null $statement, ?iterable $parameters): array
```

Comment:
I think we can use the `iterable` type for **$parameters** with a default value of an empty array. Since you pass parameters to this method with logging enabled (and they cannot be null, according to the typing of the `statement` method), or an empty array. Also, on line **703**, you use a foreach loop without checking **$parameters** for **null** before it: 
```suggestion
    protected function defineLoggerContext(
        float $queryStart, 
        PDOStatement|PDOStatementInterface|null $statement, 
        iterable $parameters = []
    ): array
```

Reply:
I've updated the method to default $parameters to an empty array and removed the nullable type hint, making it always iterable. This change allows us to skip the null check before the foreach loop, streamlining the code as per your suggestion.